javalin/src/main/java/io/javalin/websocket/WsContext.kt

```diff
@@ -26,14 +27,29 @@ abstract class WsContext(val sessionId: String, @JvmField val session: Session)
     internal val upgradeReq by lazy { session.upgradeRequest as ServletUpgradeRequest }
     internal val upgradeCtx by lazy { upgradeReq.httpServletRequest.getAttribute(upgradeContextKey) as Context }
     internal val sessionAttributes by lazy { upgradeReq.httpServletRequest.getAttribute(upgradeSessionAttrsKey) as Map<String, Any>? }
+    internal var pingFuture : ScheduledFuture<*>? = null;
+
+    companion object {
+        var executor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+    }
 
     fun matchedPath() = upgradeCtx.matchedPath
 
     fun send(message: Any) = send(upgradeCtx.jsonMapper().toJsonString(message))
     fun send(message: String) = session.remote.sendStringByFuture(message)
     fun send(message: ByteBuffer) = session.remote.sendBytesByFuture(message)
 
-    @JvmOverloads fun sendPing(message: ByteBuffer? = null) = session.remote.sendPing(message ?: ByteBuffer.allocate(0))
+    @JvmOverloads fun sendPing(applicationData: ByteBuffer? = null) = session.remote.sendPing(applicationData ?: ByteBuffer.allocate(0))
+    @JvmOverloads fun enableAutomaticPings(interval: Long = 1, unit: TimeUnit = TimeUnit.MINUTES, applicationData: ByteBuffer? = null) {
+        disableAutomaticPings();
+        pingFuture = executor.scheduleAtFixedRate({
+            sendPing(applicationData);
+        }, interval, interval, unit);
+    }
+    fun disableAutomaticPings() {
+        pingFuture?.cancel(false)
+        pingFuture = null;
```

Comment:
Hmm, what do you mean? This should work I think? pingFuture is null if there is no automatic pinging.

Reply:
I'm just wondering why you're setting it to null, shouldn't canceling it be enough?

Reply:
It's not a problem, I'm just curious.

Reply:
Ohh, that's just so that the other code now's that the future is no longer valid. I'm sure there is some sort of method to check it, but this works as well.

Reply:
Otherwise .cancel might get called twice on the same object, though I don't know if that is a problem. Doesn't hurt setting the reference to null though ^^.

Reply:
Fair enough!